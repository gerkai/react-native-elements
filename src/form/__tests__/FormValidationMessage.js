import React from 'react';
import {shallow} from 'enzyme';
import FormValidationMessage from '../FormValidationMessage';

describe('FormValidationMessage Component', () => {
  it('should render without issues', () => {
    const component = shallow(<FormValidationMessage />);

    expect(component.length).toBe(1);
    expect(component).toMatchSnapshot();
  });

  it('should show text from the FormValidationMessage children', () => {
    const component = shallow(
      <FormValidationMessage>
        There was an error
      </FormValidationMessage>
    );

    expect(component.props().children.props.children).toBe('There was an error');
  });
});
