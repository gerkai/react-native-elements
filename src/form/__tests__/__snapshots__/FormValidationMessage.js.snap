// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormValidationMessage Component should render without issues 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View
    style={
        Array [
            Object {},
            undefined,
          ]
    }
>
    <TextElement
        style={
            Array [
                Object {
                  "color": "#ff190c",
                  "fontSize": 15,
                  "marginBottom": 1,
                  "marginLeft": 20,
                  "marginRight": 20,
                  "marginTop": 5,
                },
                undefined,
                undefined,
              ]
        }
    />
</View>,
  "nodes": Array [
    <View
      style={
            Array [
                  Object {},
                  undefined,
                ]
      }
>
      <TextElement
            style={
                  Array [
                        Object {
                          "color": "#ff190c",
                          "fontSize": 15,
                          "marginBottom": 1,
                          "marginLeft": 20,
                          "marginRight": 20,
                          "marginTop": 5,
                        },
                        undefined,
                        undefined,
                      ]
            }
      />
</View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <FormValidationMessage />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <View
          style={
                    Array [
                              Object {},
                              undefined,
                            ]
          }
>
          <TextElement
                    style={
                              Array [
                                        Object {
                                          "color": "#ff190c",
                                          "fontSize": 15,
                                          "marginBottom": 1,
                                          "marginLeft": 20,
                                          "marginRight": 20,
                                          "marginTop": 5,
                                        },
                                        undefined,
                                        undefined,
                                      ]
                    }
          />
</View>,
        "_debugID": 2,
        "_renderedOutput": <View
          style={
                    Array [
                              Object {},
                              undefined,
                            ]
          }
>
          <TextElement
                    style={
                              Array [
                                        Object {
                                          "color": "#ff190c",
                                          "fontSize": 15,
                                          "marginBottom": 1,
                                          "marginLeft": 20,
                                          "marginRight": 20,
                                          "marginTop": 5,
                                        },
                                        undefined,
                                        undefined,
                                      ]
                    }
          />
</View>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <FormValidationMessage />,
}
`;
